src/librustc_typeck/astconv.rs

```diff
@@ -1815,6 +1815,16 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx> + 'o {
             hir::TyKind::Err => {
                 tcx.types.err
             }
+            hir::TyKind::CVarArgs => {
+                let va_list_did = match tcx.lang_items().va_list() {
+                    Some(did) => did,
+                    None => bug!("va_list lang item required for variadics"),
+                };
+                let env_region = tcx.mk_region(
+                    ty::ReLateBound(ty::INNERMOST, ty::BrEnv)
+                );
```

Comment:
What does this do exactly? I don't really understand the borrow checker. Will need someone else to look at this.

Reply:
I agree that we should get a second opinion here. I need to subst a lifetime here to ensure that the `VaList` cannot escape the function.

Reply:
Yeah. This may or may not be the right way to do it... I really don't know. The borrow checker is black magic.

Reply:
Yeah it may not be. I just essentially copied what we did for the `va_copy` intrinsic in [src/librustc_typeck/check/intrinsic.rs:347-365](https://github.com/rust-lang/rust/blob/master/src/librustc_typeck/check/intrinsic.rs#L347-L365).

Reply:
I'm out of my depth here, too, but I would have assumed we'd need late bound regions, so the code at least fits my expectations

Reply:
Quick update: I had considered using `ReScope` and `ReFree` here, but I was told `ReScope` was "the old way of doing things" and `ReFree` defines something at least as big as the given scope, so if the Node ID was the ID of the function body, I think that would mean the `VaList` _could_ escape the function. As a result, I think `BrAnon` is our best bet here.

Reply:
In the latest update we now create a a implicit lifetime for the spoofed `VaList` in HIR lowering. This makes the spoofed `VaList` similar to `&i32` where the lifetime is elided. The changes needed to make this happen are as follows:

 - Add a `Lifetime` to the HIR representation of `CVarArgs`.
 - Create the `Lifetime` in HIR lowering as a implicit lifetime.
 - Resolve the lifetime with `resolve_elided_lifetimes`.